src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -41,12 +41,30 @@ internal abstract class AbstractGoToDefinitionService : IGoToDefinitionService
 
         public bool TryGoToDefinition(Document document, int position, CancellationToken cancellationToken)
         {
-            // First try to compute the referenced symbol and attempt to go to definition for the symbol.
+            ISymbol symbol = null;
             var symbolService = document.GetLanguageService<IGoToDefinitionSymbolService>();
-            var (symbol, span) = symbolService.GetSymbolAndBoundSpanAsync(document, position, cancellationToken).WaitAndGetResult(cancellationToken);
-            if (symbol == null)
+
+            // If the cursor is on the keyword for overriding, we'll find the overridden symbol
+            var syntaxTree = document.GetSyntaxTreeAsync(cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
You can call GetSyntaxTreeSynchronously() rather than doing the Async and WaitAndGetResult. Much better perf. (But still prefer async whenever possible.)